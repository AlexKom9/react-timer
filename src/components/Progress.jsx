import React, { Component } from "react";
import PropTypes from "prop-types";
import "../style/progress.scss";
import bgi from "../img/bgi.jpg";

class Progress extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { name, value, percent } = this.props;
    const sqSize = 150;
    const strokeWidth = 15;
    const radius = (sqSize - strokeWidth) / 2;
    const viewBox = `0 0 ${sqSize} ${sqSize}`;
    const dashArray = radius * Math.PI * 2;
    const dashOffset = dashArray - (dashArray * percent) / 100;
    return (
      <div className="progress">
        <svg width={sqSize} height={sqSize} viewBox={viewBox}>
          <defs>
            <filter id="inset-shadow">
              <feFlood flood-color="black" />
              <feComposite operator="xor" in2="SourceGraphic" />
              <feGaussianBlur stdDeviation="2" />
              <feComposite operator="in" in2="SourceGraphic" result="map" />
              <feDiffuseLighting
                lightingColor="#fff"
                surfaceScale="4"
                diffuseConstant="1.2"
              >
                <feSpotLight x="0" y="-30" z="230" />
              </feDiffuseLighting>
              <feBlend mode="multiply" in="SourceGraphic" />
              <feComposite operator="in" in2="SourceGraphic" />
            </filter>
            <pattern
              id="bgi"
              patternUnits="userSpaceOnUse"
              width="32"
              height="32"
            >
              <image xlinkHref={bgi} width="2" height="2" />
            </pattern>
          </defs>
          <circle
            className="circle__background"
            filter="url(#inset-shadow)"
            cx={sqSize / 2}
            cy={sqSize / 2}
            r={radius}
            strokeWidth={`${strokeWidth}px`}
            // transform={`rotate(-90 ${sqSize / 2} ${sqSize / 2})`}
            fill="none"
          />
          <circle
            className={`circle__progress ${name}`}
            cx={sqSize / 2}
            cy={sqSize / 2}
            r={radius}
            strokeWidth={`${strokeWidth}px`}
            fill="transparent"
            transform={`rotate(-90 ${sqSize / 2} ${sqSize / 2})`}
            style={{
              strokeDasharray: dashArray,
              strokeDashoffset: dashOffset
            }}
          />
          <foreignObject className="node" width={sqSize} height={sqSize}>
            <body xmlns="http://www.w3.org/1999/xhtml">
              <div>
                <span className={`circle__value ${name}`}>{value}</span>
                <p className={`circle__name ${name}`}>{name}</p>
              </div>
            </body>
          </foreignObject>
        </svg>
      </div>
    );
  }
}

Progress.propTypes = {
  percent: PropTypes.number.isRequired,
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default Progress;
