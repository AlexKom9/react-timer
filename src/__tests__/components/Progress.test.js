import React from "react";
import Progress from "../../components/Progress";

import { shallow } from "enzyme";

describe(`Progress`, () => {
  test(`initial render`, () => {
    const wrapper = shallow(
      <Progress
        Progress
        name="minutes"
        value={30}
        percent={50}
      />
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
