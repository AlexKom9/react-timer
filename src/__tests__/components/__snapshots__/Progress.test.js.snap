// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress initial render 1`] = `"<div class=\\"progress\\"><svg width=\\"120\\" height=\\"120\\" viewBox=\\"0 0 120 120\\"><defs><filter id=\\"inset-shadow\\"><feFlood flood-color=\\"black\\"></feFlood><feComposite operator=\\"xor\\" in2=\\"SourceGraphic\\"></feComposite><feGaussianBlur stdDeviation=\\"2\\"></feGaussianBlur><feComposite operator=\\"in\\" in2=\\"SourceGraphic\\" result=\\"map\\"></feComposite><feDiffuseLighting lighting-color=\\"#fff\\" surfaceScale=\\"4\\" diffuseConstant=\\"1.2\\"><feSpotLight x=\\"0\\" y=\\"-30\\" z=\\"230\\"></feSpotLight></feDiffuseLighting><feBlend mode=\\"multiply\\" in=\\"SourceGraphic\\"></feBlend><feComposite operator=\\"in\\" in2=\\"SourceGraphic\\"></feComposite></filter><pattern id=\\"bgi\\" patternUnits=\\"userSpaceOnUse\\" width=\\"32\\" height=\\"32\\"><image xlink:href=\\"bgi.jpg\\" width=\\"2\\" height=\\"2\\"></image></pattern></defs><circle class=\\"circle__background\\" filter=\\"url(#inset-shadow)\\" cx=\\"60\\" cy=\\"60\\" r=\\"56\\" stroke-width=\\"8px\\" fill=\\"none\\"></circle><circle class=\\"circle__progress minutes\\" cx=\\"60\\" cy=\\"60\\" r=\\"56\\" stroke-width=\\"8px\\" fill=\\"transparent\\" transform=\\"rotate(-90 60 60)\\" style=\\"stroke-dasharray:351.85837720205683;stroke-dashoffset:175.92918860102844\\"></circle><foreignObject class=\\"node\\" width=\\"120\\" height=\\"120\\"><div><span class=\\"circle__value minutes\\">30</span><p class=\\"circle__name minutes\\">minutes</p></div></foreignObject></svg></div>"`;
